config/deploy.rb

```diff
@@ -0,0 +1,104 @@
+lock '~> 3.17.1'
+require File.expand_path('./environment', __dir__)
+
+set :use_sudo, false
+
+# If staging_case is set to "testing", capistrano deploys Commonwealth-public-interface to "testing" server.
+# switch :stage_case to "staging" when deploying Commonwealth-public-interface to staging enviroment
+# switch :stage_case to "qc" when deploying Commonwealth-public-interface to QC server
+set :stage_case, 'qc'
+# set :stage_case, 'staging'
+# set :stage_case, 'testing'
+set :application, 'Commonwealth-public-interface'
+set :repo_url, "https://github.com/boston-library/#{fetch(:application)}.git"
+set :user, Rails.application.credentials.dig("deploy_#{fetch(:stage_case)}".to_sym, :user)
+## Make user home path dynamic.
+set :deploy_to, "/home/#{fetch(:user)}/railsApps/#{fetch(:application)}"
+
+set :rvm_installed, "/home/#{fetch(:user)}/.rvm/bin/rvm"
+set :rvm_ruby_version, File.read(File.expand_path('./../.ruby-version', __dir__)).strip
+set :rvm_bundle_version, File.read(File.expand_path('./Gemfile.lock'))[-10..-1].strip
+
+# Default value for :pty is false
+set :pty, true
+
+## When running tasks against staging server, some tasks defined in it needs to be available.
+## config/deploy/staging.rb cannot be removed from <project>/shared/ directory, because it is temporarily not forcibly using ssl.
+## Otherwise "curl server_IP" returns 301....
+## As bin/puma, bin/pumactl are sensitive to current project directory, it is better not to use a symlink
+append :linked_files, 'config/database.yml', 'config/credentials/staging.key', 'config/environments/staging.rb'
+
+append :linked_dirs, 'log', 'tmp/cache', 'tmp/pids', 'tmp/sockets', 'bundle'
+
+# Default value for keep_releases is 5
+set :keep_releases, 5
+
+# Costomized tasks that restart our services
+namespace :boston_library do
+  desc 'Gem update'
+  task :gem_update do
+    on roles(:app) do
+      execute("#{fetch(:rvm_installed)} #{fetch(:rvm_ruby_version)} do gem update --system --no-document")
+    end
+  end
+
+  desc 'Install new ruby if ruby-version is required'
+  task :rvm_install_ruby do
+    on roles(:app) do
+      execute("#{fetch(:rvm_installed)} install #{fetch(:rvm_ruby_version)} -C --with-jemalloc")
+      execute("#{fetch(:rvm_installed)} use #{fetch(:rvm_ruby_version)}")
+    end
+  end
+
+  # desc 'Install bundler 2.3.26'
+  desc "Install bundler #{fetch(:rvm_bundle_version)}"
+  task :install_bundler do
+    on roles(:app) do
+      execute("#{fetch(:rvm_installed)} #{fetch(:rvm_ruby_version)} do gem install bundler:#{fetch(:rvm_bundle_version)}")
+    end
+  end
+
+  # desc 'Your bin/bundle was not generated by Bundler, so this binstub cannot run.'
+  desc 'Replace -bin/bundle- by running -bundle binstubs bundler --force-'
+  task :binstubs_bundler do
+    on roles(:app) do
+      execute("cd '#{release_path}'; #{fetch(:rvm_installed)} #{fetch(:rvm_ruby_version)} do bundle binstubs bundler --force")
+    end
+  end
```

Comment:
We didn't need this for bpldc_authority_api or ark-manager, just curious why this is needed here?

Reply:
`bin/puma` is generated by binstubs. When customized system service run bin/puma, it also check bundle's installation. If bundle is not installed using `binstubs`, it may fail to check requirements of `lockfile`, and load bundlers correctly, etc. 

After forcily running the command, it fits requirements of running `bin/puma`. Otherwise an error returns "Your bin/bundle was not generated by Bundler, so this binstub cannot run."